xformers/ops/fmha/__init__.py

```diff
@@ -321,6 +322,7 @@ def _memory_efficient_attention_forward_requires_grad(
         raise ValueError(
             f"xformers.memory_efficient_attention: Operator {op.NAME} does not support this input"
         )
+    inp.validate_bmhk()
```

Comment:
I would have seen those checks right in the beginning of the function, just before `normalize_bmhk`, this way we could also capture the fact that the tensors can be 3d or 4d